cJSON.c

```diff
@@ -479,7 +479,7 @@ static unsigned char* ensure(printbuffer * const p, size_t needed)
     }
     else
     {
-        newsize = needed * 2;
+        newsize = needed + 1;
```

Comment:
Same question here; any reason not to send this upstream and see what they say, instead of carrying a `noup` patch?

Reply:
Likewise I see the note in the PR description now. I still think we should submit it upstream and see what they say.

If upstream has a configuration mechanism for fine-tuning the library behavior, perhaps this can leverage that to make this settable in NCS.